src/modules/generateRandomNumber.js

```diff
@@ -6,8 +6,17 @@
  *
  * @return {number} A random 4-digit number
  */
+
 function generateRandomNumber() {
-  /* Write your code here */
+  const uniqRandomNumbers = new Set();
+
+  while (uniqRandomNumbers.size !== 4) {
+    const randomNumber = Math.floor(1 + Math.random() * 9);
+
+    uniqRandomNumbers.add(randomNumber);
+  }
+
+  return Number(Array.from(uniqRandomNumbers).join(''));
```

Comment:
The conversion of the set to a number may result in fewer than 4 digits if the set contains leading zeros. Ensure that the number is always 4 digits long by adjusting the logic to handle leading zeros appropriately.